File: 1653-minimum-deletions-to-make-string-balanced/1653-minimum-deletions-to-make-string-balanced.java
class Solution {
    public int minimumDeletions(String s) {
        int count = 0;
        int del = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            char ch = s.charAt(i);
            if (ch == 'a') {
                count++;
            } else {
                if (count > 0) {
                    del++;
                    count--;
                }
            }
        }
        return del;
    }
}

File: 1653-minimum-deletions-to-make-string-balanced/1653-minimum-deletions-to-make-string-balanced.cpp
class Solution {
public:
    int minimumDeletions(string s) {
      int count=0;
      int del=0;
      for (int i = s.size() - 1; i >= 0; i--) {
            if (s[i] == 'a') {
                count++;
            } 
             else if (s[i] == 'b' && count> 0) {
                del++;
                count--;
            }
        }
        return del;
    }
};